*** BasicBot.pm	Sun Jan 26 13:54:55 2003
--- /Library/Perl/Bot/BasicBot.pm	Thu Mar 13 03:54:35 2003
***************
*** 292,298 ****
                                    args    => {
                                                channel => $args->{channel},
                                                who     => $args->{who},
!                                               address => $args->{address}
                                               }
                                   };
    return undef;
--- 292,299 ----
                                    args    => {
                                                channel => $args->{channel},
                                                who     => $args->{who},
!                                               address => $args->{address},
!                                               data => $args->{data},
                                               }
                                   };
    return undef;
***************
*** 822,827 ****
--- 823,830 ----
    # work out who it was from
    $mess->{who} = $this->nick_strip($nick);
  
+   return undef if $this->ignore_nick($mess->{who});
+ 
    # right, get the list of places this message was
    # sent to and work out the first one that we're
    # either a memeber of is is our nick.
***************
*** 830,836 ****
    # /does/ this, so we can get away with this:
  
    my $channel = $to->[0];
!   if ($channel eq $this->nick)
    {
      $mess->{channel} = "msg";
      $mess->{address} = "msg";
--- 833,839 ----
    # /does/ this, so we can get away with this:
  
    my $channel = $to->[0];
!   if ($this->nick =~ /^$channel$/i)
    {
      $mess->{channel} = "msg";
      $mess->{address} = "msg";
***************
*** 845,857 ****
    unless ($mess->{channel} eq "msg")
    {
      my $nick = $this->nick;
!     ($mess->{address}) = $mess->{body} =~ s/^(\Q$nick\E\s*[:,-]?)//;
  
      foreach $nick ($this->alt_nicks)
      {
!       last if $this->{address};
  
!       ($mess->{address}) = $mess->{body} =~ s/^(\Q$nick\E\s*[:,-])//;
      }
    }
  
--- 848,862 ----
    unless ($mess->{channel} eq "msg")
    {
      my $nick = $this->nick;
!     ($mess->{address}) = $mess->{body} =~ /^(\Q$nick\E\s*[:,-])/i;
!     $mess->{body} =~ s/^(\Q$nick\E\s*[:,-])//i if $mess->{address};
  
      foreach $nick ($this->alt_nicks)
      {
!       last if $mess->{address};
  
!       ($mess->{address}) = $mess->{body} =~ /^(\Q$nick\E\s*[:,-])/i;
!       $mess->{body} =~ s/^(\Q$nick\E\s*[:,-])//i if $mess->{address};
      }
    }
  
***************
*** 861,882 ****
  
    # okay, we got this far.  Better log this.  This needs changing
    # to a nice format.  Oooh, I could spit out sax events... (mf)
!   $this->log((join '||', $mess->{who},
!                          $mess->{channel},
!                          $mess->{address},
!                          $mess->{body})."\n");
  
    # check if someone was asking for help
!   if ($mess->{address} && ($mess->{body} =~ /^help/i))
!   {
!     $this->log("Invoking help for '$mess->{who}'\n");
!     $mess->{body} = $this->help($mess);
!     $this->say($mess);
!     return;
!   }
  
    # okay, call the said/emoted method
!   my $respond = $this->$received($mess);
  
    ### what did we get back?
  
--- 866,889 ----
  
    # okay, we got this far.  Better log this.  This needs changing
    # to a nice format.  Oooh, I could spit out sax events... (mf)
!   if ($mess->{channel} eq 'msg') {
!     $this->log("/msg ${\($this->nick)} $mess->{body} (from $mess->{who})\n");
!   }
!   else {
!     $this->log("$mess->{who}:${\($mess->{address} ? ' ' . $mess->{address} : '')} $mess->{body} (from $mess->{channel})\n");
!   }
  
    # check if someone was asking for help
!   #if ($mess->{address} && ($mess->{body} =~ /^help/i))
!   #{
!   #  $this->log("Invoking help for '$mess->{who}'\n");
!   #  $mess->{body} = $this->help($mess);
!   #  $this->say($mess);
!   #  return;
!   #}
  
    # okay, call the said/emoted method
!   $respond = $this->$received($mess);
  
    ### what did we get back?
  
***************
*** 1037,1044 ****
    my $this = shift;
    my $nick = shift;
  
!   return 1 if $nick =~ /bot$/;
!   return grep {$nick eq $_} @{ $this->{ignore_list} };
  }
  
  =item nick_strip
--- 1044,1051 ----
    my $this = shift;
    my $nick = shift;
  
!   return 1 if $nick =~ /bot$/i;
!   return grep {$nick =~ /^$_$/i} @{ $this->{ignore_list} };
  }
  
  =item nick_strip
