*** IRC.pm	Sun Feb  2 18:23:21 2003
--- /Library/Perl/POE/Component/IRC.pm	Mon Mar 10 01:00:56 2003
***************
*** 64,70 ****
--- 64,72 ----
      'squit'     => [ PRI_HIGH,   \&onlytwoargs,   ],
      'kill'      => [ PRI_HIGH,   \&onlytwoargs,   ],
      'privmsg'   => [ PRI_NORMAL, \&privandnotice, ],
+     'privmsglow' => [ PRI_HIGH+5,   \&privandnotice, ],
      'notice'    => [ PRI_NORMAL, \&privandnotice, ],
+     'noticelow'  => [ PRI_HIGH+5,   \&privandnotice, ],
      'join'      => [ PRI_HIGH,   \&oneortwo,      ],
      'summon'    => [ PRI_HIGH,   \&oneortwo,      ],
      'sconnect'  => [ PRI_HIGH,   \&oneandtwoopt,  ],
***************
*** 790,795 ****
--- 792,800 ----
    my ($kernel, $state, $to) = @_[KERNEL, STATE, ARG0];
    my $message = join ' ', @_[ARG1 .. $#_];
    my $pri = $irc_commands{$state}->[CMD_PRI];
+ 
+   $state =~ s/privmsglow/privmsg/;
+   $state =~ s/noticelow/notice/;
  
    unless (defined $to and defined $message) {
      die "The POE::Component::IRC event \"$state\" requires two arguments";
